### ISSUE TITLE
Enforce code style

### ISSUE THREAD

**Issue by MartijnHols** [MartijnHols/WoWAnalyzer#64]
Whenever reviewing PRs I see a bunch of code style choices that bug me, but I usually figure it's pointless to argue requiring changes just because of that. In order to reduce the inconsistencies I'd like to enforce some more eslint rules;

- [x] prefer-const
- [x] no-var
- [x] react/prop-types
- [x] one-var
- [x] order of imports (global/local)
- [x] require semicolons

I looked into using `Prettier` but it's pretty opinionated in the wrong way. For example it would reformat stuff to this:
```
const something =
  <div>Hello</div>;
```
In the first place this should be single line and then it would be ok. But if it was too long because for example the JSX was much longer, then it should be wrapped in parenthesis:
```
const something = (
  <div>Hello</div>
);
```
The reason for this is that it is consistent with requiring parenthesis when returning JSX;
```
return (
  <div>Hello</div>
);
```
and it makes sense linguistically (you wouldn't put a comma on the next line) and is easier to follow; when I see `const something = (` I know the line is to be continued on the next line. Without parenthesis the JSX is just floating somewhere. The same goes for this:
```
{condition &&
  <div>
    Hello!
  </div>}
```
I looked into prettier-eslint, but ESLint is currently unable to add back the parenthesis after Prettier removed it, so that doesn't solve it. Ternaries are also messy.
Any solution with Prettier seems to be sub-optimal because of these wrongly opinionated issues, so even though CRA suggests using it instead of stricter ESLint rules, I reckon it is not a viable alternative at this time.

This will probably require ejecting, but I did that in the past and it's not that big a deal. The nice thing is that I'll also be able to remove `.env` again since the docs for that state it's recommended not to commit it.

**Comment by MartijnHols** [comment]
Not using Airbnb style because it is a bit too extreme. In addition is lists everything as `error` which would interfere with compile time checking.

**Comment by MartijnHols** [comment]
I added a bunch of rules that should help with the more basic issues. I decided against adding everything in the Airbnb rules to prevent the rules from becoming too annoying.

`comma-dangle` rules make the comma usage consistent. From what I saw the biggest *offenders* were also inconsistent in their usage. This makes commits cleaner and makes it very slightly easier to add new lines to arrays.
Semicolons: these are required.
Vars: don't use `var`, it's never needed and may cause issues, `let` is a perfectly fine replacement without downsides. `prefer-const` suggests using `const` when possible, making code easier to follow.
PropTypes: it's easy to forget sometimes, these rules should help keep them right.
Imports: without an eslint rule this would become anarchy. The enable rule should add somewhat of a structure to it.
